QuipEditor.py

```diff
@@ -171,6 +172,21 @@ def on_pre_close(self, view):
 		manager.remove_tab(view=view)
 
 
+class ShowCommentsOnHover(sublime_plugin.EventListener):
+
+	def on_hover(self, view, point, hover_zone):
+		thread_id = manager.get_thread(view)
+		if not thread_id or view.is_popup_visible():
+			return
+		word_region = view.word(point)
+		word = view.substr(word_region)
+		comments = [str(comment) for comment in manager.comments.get(thread_id) if word in comment.sections]
+		view.sel().clear()
+		view.sel().add(Region(point))
+		view.show_popup_menu(comments, None)
```

Comment:
when popup will be hidden?

Reply:
We dont need to explicitly close the popup. Sublime will take all the logic itself

The `popup_menu` will be closed when you click anywhere except this popup (anything else will be frozen until that anyway)
The `popup` have few triggers (flags) to be closed (from the documentation):
- sublime.COOPERATE_WITH_AUTO_COMPLETE. Causes the popup to display next to the auto complete menu
- sublime.HIDE_ON_MOUSE_MOVE. Causes the popup to hide when the mouse is moved, clicked or scrolled
- sublime.HIDE_ON_MOUSE_MOVE_AWAY. Causes the popup to hide when the mouse is moved (unless towards the popup), or when clicked or scrolled

In my opinion, `sublime.HIDE_ON_MOUSE_MOVE_AWAY` is the best way to close it, because you can move you cursor on the popup itself (and it feels more natural), rather than it will be closed immediately when you moved you cursor from the point it was triggered